piq/vif.py

```diff
@@ -46,8 +46,8 @@ def vif_p(x: torch.Tensor, y: torch.Tensor, sigma_n_sq: float = 2.0,
     _validate_input([x, y], dim_range=(4, 4), data_range=(0, data_range))
 
     min_size = 41
-    if x.size(-1) < min_size or x.size(-2) < min_size:
-        raise ValueError(f'Invalid size of the input images, expected at least {min_size}x{min_size}.')
+    assert x.size(-1) >= min_size and x.size(-2) >= min_size, \
+        f'Invalid size of the input images, expected at least {min_size}x{min_size}.'
 
```

Comment:
I think this one should remain an exception. Disabling it with python -O makes no sense, the code will fail later anyway. What we want to do with rising this exception is to make it fail early and save compute time.

Reply:
Let's discuss it in this tread :).

We do use exceptions to raise error at earliest stage. However, we decided to pursue to approach with assertion errors instead of the usual exceptions to allow "pro" users turn off validation for faster processing. As we stated in readme, `-O` option could be used at user's risk with less information for debugging. As a result, we shouldn't worry about detecting the error at earliest stage with `-O` flag used.

Moreover, `kernel_size` parameter verification is an addition to input tensor validation `_validate_input`, which uses `assert`s.

Due to these reasons, I would suggest to use `assert` for validation to support coherence between other validations and our description of `python -O` option.
@snk4tr, please, let me know your feedback.